### ISSUE TITLE
Fix accessibility linting violations

### ISSUE THREAD

**Issue by sampart** [github/vscode-codeql#2560]
Once https://github.com/github/vscode-codeql/pull/2561 is merged, the following linting rules will be enabled:

```
jsx-a11y/anchor-is-valid
jsx-a11y/no-noninteractive-element-interactions
jsx-a11y/no-static-element-interactions
jsx-a11y/click-events-have-key-events
```

Linter errors for existing violations of these rules have been suppressed with `eslint-disable-next-line`, but should be fixed.

You can find the violations by searching the codebase for `jsx-a11y`, which will show you the disablements.

**Comment by robertbrignull** [comment]
See https://github.com/github/vscode-codeql/pull/2561 for the locations of disabled linting rules. Or search for all cases of `eslint-disable-next-line` and filter to those mentioning `jsx-a11y` rules.

**Comment by robertbrignull** [comment]
From a quick glance I did yesterday, I believe that all of the violations can be fixed without high effort. If that turns out to be wrong, we can decide to not fix these violations and leave them as they are. Or at least we can then make a decision about how bad the violation is and if necessary schedule more time for it.

**Comment by robertbrignull** [comment]
I had a brief look at actually fixing these and I now not sure it's so easy. For example for the links in the results view we want them to look like links but we currently aren't (maybe can't) provide a href so we use a `onClick` handler. Our options are to either provide a working href or replace them with buttons styled to look like links. Unless there are other options I'm unaware of.

**Comment by koesie10** [comment]
PR for fixing some of these violations: https://github.com/github/vscode-codeql/pull/2719

**Comment by koesie10** [comment]
PR for fixing the remaining violations: https://github.com/github/vscode-codeql/pull/2726

**Comment by koesie10** [comment]
Both PRs have been merged and there are no more `jsx-a11y/*` violations
